include/boost/math/tools/simple_continued_fraction.hpp

```diff
@@ -171,6 +208,41 @@ std::ostream& operator<<(std::ostream& out, simple_continued_fraction<Real, Z2>&
    return out;
 }
 
+template<typename Real, typename Z = std::int64_t>
+inline auto simple_continued_fraction_coefficients(Real x)
+{
+    auto temp = simple_continued_fraction<Real, Z>(x);
+    return temp.get_data();
+}
+
+// Can be used with `boost::rational` from <boost/rational.hpp>
+template <typename Rational, typename Real, typename Z = std::int64_t>
+inline Rational to_rational(const simple_continued_fraction<Real, Z>& scf)
```

Comment:
Does this work with other rational types [(e.g. `mpq_t`)](https://gmplib.org/manual/Rational-Number-Functions)?

Reply:
I assume that `mpq_t` cannot be used since it is C-style struct which requires to call `mpq_init` etc.
The function can be used with any type `T` that supports `typename T::int_type`, `T(typename T::int_type)` (just integer part) and `T(typename T::int_type, typename T::int_type)` (numerator and denominator).
I assume C++20 is not supported so I do not use concepts or such.